### PR TITLE
[code hashtable] change remove function so that it does not fail on removing a non-existing element

### PR DESCRIPTION
**Pull request by lewurm**
saves us one lookup operation

### CODE REVIEW

File: mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -2483,8 +2483,7 @@ mono_jit_free_method (MonoDomain *domain, MonoMethod *method)
 	if (mono_use_interpreter) {
 		mono_domain_jit_code_hash_lock (domain);
 		/* InterpMethod is allocated in the domain mempool */
-		if (mono_internal_hash_table_lookup (&info->interp_code_hash, method))
-			mono_internal_hash_table_remove (&info->interp_code_hash, method);
+		mono_internal_hash_table_remove (&info->interp_code_hash, method);
```
Comment (kumpera) [inline mono/mini/mini-runtime.c]: Shouldn't we add a g_assert to the other call to mono_internal_hash_table_remove?

Comment (lewurm) [reply mono/mini/mini-runtime.c]: fair enough. updated!

**Comment by lewurm** [comment]
@monojenkins squash

File: mono/mini/mini-runtime.c [hunk 2]
```diff
@@ -2482,9 +2482,9 @@ mono_jit_free_method (MonoDomain *domain, MonoMethod *method)
 
 	if (mono_use_interpreter) {
 		mono_domain_jit_code_hash_lock (domain);
-		/* InterpMethod is allocated in the domain mempool */
-		if (mono_internal_hash_table_lookup (&info->interp_code_hash, method))
-			mono_internal_hash_table_remove (&info->interp_code_hash, method);
+		/* InterpMethod is allocated in the domain mempool. We might haven't
```
Comment (luhenry) [inline mono/mini/mini-runtime.c]: `We might not have`

**Pull request merged**